tools/reeve/tests/test_signals.py

```
@@ -0,0 +1,113 @@
+    other = {"schema": 1, "kind": "something-else"}
+    text = json.dumps(other) + "\n" + json.dumps(_rec()) + "\n"
+    records = signals.parse_log(text)
+    assert len(records) == 1 and records[0]["kind"] == "gate-run"
+
+
+def test_parse_log_empty_is_no_history():
+    assert signals.parse_log("") == []
+
+
+def test_parse_log_malformed_line_raises_with_lineno():
+    with pytest.raises(ValueError, match="log.ndjson:2: not valid JSON"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)